### ISSUE TITLE
Wesnoth terminates if an error occurs during editor reload

### ISSUE THREAD

**Issue by CelticMinstrel** [wesnoth/wesnoth#3391]
If you reload using F5 while in the editor, and a WML error occurs, Wesnoth closes. It should instead either dump you to the main menu or offer an option to try again after you've fixed the issue.

**Comment by Pentarctagon** [comment]
@CelticMinstrel is this still the case after #6702?

**Comment by CelticMinstrel** [comment]
I'll check, uhhh… soon? Please remind me if I don't get back to this in a few days.

**Comment by Pentarctagon** [comment]
*reminds*

**Comment by CelticMinstrel** [comment]
Yes, it still crashes if it encounters malformed WML. It does offer the usual "without addons" retry though. I only tried introducing an error in core WML (editor-specific stuff). I assume it doesn't crash if the error is in an add-on.

It's particularly bad to crash in this case because there could be unsaved work. I think it would be better if the WML error dialog had "Retry" and "Quit" buttons, or something like that.
